capstone/capdb/models.py

```diff
@@ -979,17 +987,23 @@ def get_frontend_url(self, cite=None, disambiguate=False, include_host=True):
             If disambiguate is true, return /series/volnum/pagenum/id/.
         """
         cite = cite or self.citations.first()
-        # try to match "(volnumber) (series) (pagenumber)"
-        m = re.match(r'(\S+)\s+(.+?)\s+(\S+)$', cite.cite)
+
+        # try to match "(volnumber) (Series) (pagenumber)"
+        m = re.match(r'^(\S+)\s+([A-Z].+?)\s+(\S+)$', cite.cite)
+
         if not m:
             # if cite doesn't match the expected format, always disambiguate so URL resolution doesn't depend on cite value
             disambiguate = True
             # try to match "(year)-(series)-(case index)", e.g. "2017-Ohio-5699" and "2015-NMCA-053"
             m = re.match(r'(\S+)-(.+?)-(\S+)$', cite.cite)
         # TODO: final fallback value is wrong, because first_page is the physical page count and not the page label
         # after token streams are in, we should be able to retrieve the actual page label instead
-        cite_parts = m.groups() if m else [self.volume.volume_number, self.reporter.short_name, self.first_page]
-        args = [slugify(cite_parts[1]), cite_parts[0], cite_parts[2]] + ([self.id] if disambiguate else [])
+        volume_number, rep_short_nm, fp = m.groups() if m else [self.volume.volume_number_slug, self.reporter.short_name, self.first_page]
+
+        if self.volume.volume_number_slug:
+            volume_number = self.volume.volume_number_slug
```

Comment:
Maybe cleaner just to use `slugify(volume_number)` below instead of using self.volume.volume_number_slug? This current code assumes that the citation we're using to build the frontend_url will always match the book volume, which I *think* is true but maybe isn't necessary to assume.

Reply:
Done